### ISSUE TITLE
Mark `enquirer` as a optional peer dependency

### ISSUE THREAD

**Issue by micalevisk** [cenk1cenk2/listr2#589]
Hi!

I've been using https://github.com/okonet/lint-staged CLI and they have `listr2` as a dependency. But they don't need `enquirer` (for whatever reason), so `lint-staged` users will get **warnings** like `"lint-staged > listr2@3.13.3" has unmet peer dependency "enquirer@>= 2.3.0 < 3"`.  
I do know we can ignore those warnings tho

But maybe you can mark this peer dependency as optional as `listr2` will work as expected (for some use cases) without it?

https://github.com/cenk1cenk2/listr2/blob/bd5a4a08c5c441d41ad1bb1f2344226b980879fd/package.json#L101-L103

basically adding [`peerDependenciesMeta`](https://docs.npmjs.com/cli/v7/configuring-npm/package-json#peerdependenciesmeta) entry on `package.json`, like:

```json
  "peerDependenciesMeta": {
    "enquirer": {
      "optional": true
    }
  }
```

PS: I didn't have used `listr2` package thus I'm not if not emitting that warning is fine.

**Comment by cenk1cenk2** [comment]
Hello @micalevisk ,
That would be totally fine to add so I will do that. Thanks for the heads up.

It will work fine as being not installed as long as the underlying package does not use the prompts since it is dynamically imported.
